go/vt/orchestrator/logic/tablet_discovery.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logic
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"vitess.io/vitess/go/vt/orchestrator/db"
+	"vitess.io/vitess/go/vt/orchestrator/external/golib/log"
+	"vitess.io/vitess/go/vt/orchestrator/external/golib/sqlutils"
+	"vitess.io/vitess/go/vt/orchestrator/inst"
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+	"vitess.io/vitess/go/vt/topo"
+	"vitess.io/vitess/go/vt/topotools"
+	"vitess.io/vitess/go/vt/vttablet/tmclient"
+)
+
+var (
+	ts *topo.Server
+)
+
+// OpenTabletDiscovery opens the vitess topo if enables and returns a ticker
+// channel for polling.
+func OpenTabletDiscovery() <-chan time.Time {
+	// TODO(sougou): If there's a shutdown signal, we have to close the topo.
+	ts = topo.Open()
+	// TODO(sougou): remove ts and push some functions into inst.
+	inst.TopoServ = ts
+	// Clear existing cache and perform a new refresh.
+	if _, err := db.ExecOrchestrator("delete from vitess_tablet"); err != nil {
```

Comment:
I don't like how there's a substantial(?) time where there's no `vitess_tablet` info, which then implies no `analysis` info.

May I suggest doing the change in a transaction? I realize the iteration on cells->tablets is not transaction friendly. How about if we create a table called `vitess_tablet_scratch`, we `delete` from it, we `insert` (SaveTablet) to it, and once we're done, we can:
```sql
begin;
delete from vitess_vttablet;
insert into vitess_vttablet select * from vitess_vttablet_scrap;
commit;
```
?

Reply:
The initialization has confused me. So, I'll probably need a crash course from you.

The main reason doing this here is to trigger a `ReadTopologyInstance` for each tablet that's discovered. However, calling it at this time did not always succeed because the Orc sometimes comes up as non-leader, and if so, fails all these calls.

On the other hand, it also waits 15 seconds before starting analysis, by which time all this is already loaded and ready to go.

Ideally, we should load this up in one sweep, read all the topologies, and immediately start running analysis. But I couldn't make this happen correctly. So, I left this the way it is for now.

Reply:
Resolved in 1:1: call `InjectSeed()` function to populate instances upon startup.